src/nfts/ERC1155AdminMinter.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {Roles} from "@protocol/core/Roles.sol";
+import {CoreRef} from "@protocol/refs/CoreRef.sol";
+import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.sol";
+
+contract ERC1155AdminMinter is CoreRef {
+    /// --------- Structs ---------- ///
+
+    struct BulkMint {
+        address nftContract;
+        uint256 tokenId;
+        uint256 amount;
+        address recipient;
+    }
+
+    /// --------- Events ---------- ///
+
+    /// @notice - Event emitted when the admin mint is successful
+    event TokensMinted(
+        address indexed nftContract,
+        address indexed recipient,
+        uint256 amount,
+        uint256 tokenId
+    );
+
+    constructor(address _core) CoreRef(_core) {}
+
+    /// @notice mint ERC1155 tokens as admin
+    /// @param nftContract the token contract to mint from
+    /// @param tokenId the id of the token to mint
+    /// @param amount the amounts of the token to buy
+    /// @param recipient the address to send the ERC11-55 tokens
+    /// @dev locks up to level 1
+    function mintToken(
+        address nftContract,
+        address recipient,
+        uint256 tokenId,
+        uint256 amount
+    ) external onlyRole(Roles.ADMIN) globalLock(1) {
```

Comment:
We should also add the modifiers `whenNotPaused onlyWhitelist(nftContract)`.

Reply:
I would keep the pause modifier as the underlying functions will still pause and tx's wont go through. At least having the modifier tell us this by looking at the function signature. Ie this still works: https://github.com/ZTX-Foundation/tuxedo/pull/59/files#diff-091e511200b6c4bb44135de54f77bb5328f79543212395e77232568d179973beR55